### ISSUE TITLE
Use Async instead of forcing to synchronous execution

### ISSUE THREAD

**Issue by StefH** [JDetmar/NLog.Extensions.AzureStorage#23]
In the table storage code, I see this:
``` c#
protected override void Write(LogEventInfo logEvent)
```

And later:
``` c#
private static void TableExecute(CloudTable cloudTable, TableOperation insertOperation)
        {
#if NETSTANDARD
            cloudTable.ExecuteAsync(insertOperation).GetAwaiter().GetResult();
#else
            cloudTable.Execute(insertOperation);
#endif
}
```

Why not use the **WriteAsyncLogEvent()** and other async methods ?

**Comment by snakefoot** [comment]
When NLog 4.6 gets out, then `AsyncTaskTarget` will be able to handle this (Support batching + timeout-handling + retry).

**Comment by brolaugh** [comment]
NLog 4.6 is has been out for a while now. What's the status of this?

**Comment by snakefoot** [comment]
Yes what is the world coming to! where coding monkeys for free are not doing their jobs :)

**Comment by snakefoot** [comment]
@brolaugh & @StefH Initial attempt to implemented here:  #43 (Code review is welcome)

**Comment by snakefoot** [comment]
@brolaugh & @StefH Please try these updated packages:

> [NLog.Extensions.AzureCosmosTable](https://www.nuget.org/packages/NLog.Extensions.AzureCosmosTable/)
> [NLog.Extensions.AzureBlobStorage](https://www.nuget.org/packages/NLog.Extensions.AzureBlobStorage/)
> [NLog.Extensions.AzureQueueStorage](https://www.nuget.org/packages/NLog.Extensions.AzureQueueStorage/)
> [NLog.Extensions.AzureEventHub](https://www.nuget.org/packages/NLog.Extensions.AzureEventHub/)

And report your findings.
